com.sap.cloud.lm.sl.cf.core/src/main/java/com/sap/cloud/lm/sl/cf/core/cf/metadata/processor/MtaMetadataParser.java

```diff
@@ -0,0 +1,100 @@
+package com.sap.cloud.lm.sl.cf.core.cf.metadata.processor;
+
+import static com.sap.cloud.lm.sl.cf.core.cf.metadata.util.MtaMetadataUtil.getMtaId;
+import static com.sap.cloud.lm.sl.cf.core.cf.metadata.util.MtaMetadataUtil.getMtaVersion;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import org.cloudfoundry.client.lib.domain.CloudApplication;
+import org.cloudfoundry.client.lib.domain.CloudEntity;
+import org.cloudfoundry.client.lib.domain.CloudService;
+import org.cloudfoundry.client.v3.Metadata;
+import org.springframework.stereotype.Component;
+
+import com.sap.cloud.lm.sl.cf.core.cf.metadata.ImmutableMtaMetadata;
+import com.sap.cloud.lm.sl.cf.core.cf.metadata.MtaMetadata;
+import com.sap.cloud.lm.sl.cf.core.cf.metadata.MtaMetadataAnnotations;
+import com.sap.cloud.lm.sl.cf.core.cf.metadata.MtaMetadataLabels;
+import com.sap.cloud.lm.sl.cf.core.message.Messages;
+import com.sap.cloud.lm.sl.cf.core.model.DeployedMtaModule;
+import com.sap.cloud.lm.sl.cf.core.model.DeployedMtaResource;
+import com.sap.cloud.lm.sl.common.ParsingException;
+import com.sap.cloud.lm.sl.common.util.JsonUtil;
+
+@Component
+public class MtaMetadataParser {
+
+    private static final List<String> METADATA_LABELS = Arrays.asList(MtaMetadataLabels.MTA_ID, MtaMetadataLabels.MTA_VERSION);
+
+    public boolean hasMtaMetadata(CloudEntity entity) {
+        Metadata metadata = entity.getV3Metadata();
+        if (metadata == null || metadata.getLabels() == null) {
+            return false;
+        }
+        return Collections.disjoint(metadata.getLabels()
```

Comment:
Shouldn't this be negated?

Reply:
It should. Will replace this with stream anyMatch to avoid mistakes as this one and to avoid the complexity.